internal/rpcnetworkclient/client.go

```diff
@@ -0,0 +1,219 @@
+			diagnosticEventsXDR: response.DiagnosticEventsXDR,
+		}
+	}
+
+	return r.pollTransactionStatus(ctx, response.Hash)
+}
+
+// pollTransactionStatus polls GetTransaction until the transaction is finalized
+// (SUCCESS or FAILED) or the context times out.
+func (r *NetworkClient) pollTransactionStatus(ctx context.Context, txHash string) error {
+	b := backoff.NewExponentialBackOff()
+	b.InitialInterval = backoffInitialInterval
+	b.MaxInterval = backoffMaxInterval
+	// Note: MaxElapsedTime is not set here because the context timeout
+	// (submitTransactionTimeout) already handles the overall timeout.
+
+	var finalErr error
+	err := backoff.Retry(func() error {
+		txResponse, err := r.client.GetTransaction(ctx, protocol.GetTransactionRequest{
+			Hash: txHash,
+		})
+		if err != nil {
+			finalErr = &NetworkError{err: err}
+			return backoff.Permanent(finalErr)
+		}
+
+		switch txResponse.Status {
+		case protocol.TransactionStatusSuccess:
+			return nil
+		case protocol.TransactionStatusFailed:
+			finalErr = &NetworkError{
+				err:                 fmt.Errorf("transaction failed"),
+				resultXDR:           txResponse.ResultXDR,
+				diagnosticEventsXDR: txResponse.DiagnosticEventsXDR,
+			}
+			return backoff.Permanent(finalErr)
+		default:
+			// Transaction not yet processed (including NOT_FOUND status).
+			// After sendTransaction returns a hash, the transaction should
+			// eventually appear, so we retry until context timeout.
+			return fmt.Errorf("transaction not yet finalized")
+		}
+	}, backoff.WithContext(b, ctx))
+
+	if err != nil {
+		if finalErr != nil {
+			return finalErr
+		}
+		return &NetworkError{err: fmt.Errorf("timeout waiting for transaction %s to finalize", txHash), timeout: true}
+	}
+	return nil
+}
+
+// GetAccountDetails retrieves account details using the underlying RPC client.
+func (r *NetworkClient) GetAccountDetails(accountID string) (*internal.AccountDetails, error) {
+	accountIDObj, err := xdr.AddressToAccountId(accountID)
+	if err != nil {
+		return nil, &NetworkError{err: err}
+	}
+
+	ledgerKey, err := accountIDObj.LedgerKey()
+	if err != nil {
+		return nil, &NetworkError{err: err}
+	}
+
+	ledgerKeyXDR, err := xdr.MarshalBase64(ledgerKey)
+	if err != nil {
+		return nil, &NetworkError{err: err}
+	}
+
+	resp, err := r.client.GetLedgerEntries(context.Background(), protocol.GetLedgerEntriesRequest{
+		Keys: []string{ledgerKeyXDR},
+	})
+	if err != nil {
+		return nil, &NetworkError{err: err}
+	}
+
+	if len(resp.Entries) != 1 {
+		return nil, &NetworkError{
+			err:      fmt.Errorf("failed to find ledger entry for account %s", accountID),
+			notFound: true,
+		}
+	}
+
+	var entry xdr.LedgerEntryData
+	if err := xdr.SafeUnmarshalBase64(resp.Entries[0].DataXDR, &entry); err != nil {
+		return nil, &NetworkError{err: err}
+	}
+	if entry.Type != xdr.LedgerEntryTypeAccount {
+		return nil, &NetworkError{err: fmt.Errorf("unexpected ledger entry type: expected account, got %v", entry.Type)}
```

Comment:
this should never _ever_ happen, but good to be defensive here since we'd want to know that